docs/ancestry.md

```diff
@@ -1264,14 +1265,96 @@ print(tree.tmrca(0,1))
+Msprime provides the option to perform coalescent approximations
+to selective sweeps, in which a beneficial mutation moves through
+the population. This is done through the use of a structured 
+coalescent model in the spirit of 
+[Braverman et al. (1995)](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC1206652/)
+Please see the {class}`API documentation<.SweepGenicSelection>`
+for a formal description of the model.
 
-```{eval-rst}
-.. todo:: examples of the selective sweeps models. We want to have
-    a single sweep reverting to Hudson, and also lots of sweeps.
-```
+Looking backwards in time the population enters a sweep phase where
+the population is split between two classes of selective backgrounds,
+those linked to the beneficial allele, call it {math}`B`, and those not,
+{math}`b`. During this portion of the simulation we track competing
+rates of coalescence and recombination on the {math}`B` and {math}`b`
+backgrounds.
+
+This implementation is reasonable general, althougth there are 
+current limitations (e.g., no change of population size during
+a sweep). The user supplies a final allele frequency and a starting
+allele frequency, between which msprime will simulate a stochastic 
+sweep trajectory according to a conditional diffusion model [(Coop
+and Griffiths, 2004](https://pubmed.ncbi.nlm.nih.gov/15465123/);
+[Kim and Stephan 2002](https://pubmed.ncbi.nlm.nih.gov/11861577/)).
+
+{math}`\alpha`
+Beyond the start and end frequencies of the sweep trajectory, the user
+must specific the strength of selection as measured by 
+{math}`\alpha = 2N_es` where s is the selective
+advantage of the {math}`B` homozygote over the {math}`b` homozygote
+and {math}`h=0.5`. The position represents the location along
+the chromosome where the beneficial allele occurs.
+ All other parameters can be set as usual. 
+
+As an example, let perform 50 replicate hard sweep simulations using 
+msprime and then plot mean {math}`\pi` in windows across the simulated
+region. The position is set to the middle of the chromosome, so we
+expect to see the characteristic valley of diversity following our sweeps.
+The key step here is to build a list of models which includes
+a `SweepGenicSelection` model.
+
+```{code-cell}
+from matplotlib import pyplot as plt
+
+sample_size = 10
+nreps = 50
+seq_length = 1e6
+r = 1e-7
+mu = 1e-8
+Ne = 1e3
+s = 0.25
+alpha = 2 * Ne * s 
+tau = 0 
+
+# define hard sweep model
+mod = msprime.SweepGenicSelection(
+    position=seq_length / 2,
+    start_frequency=1.0 / (2 * Ne),
+    end_frequency=1.0 - (1.0 / (2 * Ne)),
+    alpha=alpha,
+    dt=1e-6,
+)
+
+# make model list to hand to simulate
+mod_list = [None]
+mod_list.append((tau, mod))
+mod_list.append((None, 'hudson'))
+
+reps = msprime.simulate(
+    sample_size,
+    Ne=Ne,
+    model=mod_list,
+    recombination_rate=r,
+    length = seq_length,
+    num_replicates=nreps,
+    mutation_rate=mu,
+    )
+
+wins = np.array(list(range(0,int(seq_length+1),int(seq_length/20))))
```

Comment:
np.linspace is better for this https://numpy.org/doc/stable/reference/generated/numpy.linspace.html